src/dodal/utils.py

```diff
@@ -249,9 +249,9 @@ def collect_factories(
         if (
             callable(var)
             and is_any_device_factory(var)
-            and (include_skipped or not _is_device_skipped(var))
+            and (include_skipped or not _is_device_skipped(var))  # type: ignore
         ):
-            factories[var.__name__] = var
+            factories[var.__name__] = var  # type: ignore
```

Comment:
Should: Thanks for fixing this but I think https://github.com/DiamondLightSource/dodal/pull/783 is the better fix, could you review and merge that in and revert this?